spec/jobs/inherit_permissions_job_spec.rb

```diff
@@ -2,92 +2,199 @@
-          expect(file.edit_users).to match_array [user.to_s, "abc@123.com"]
+          expect(file.read_groups).to match_array ["my_read_group"]
+          expect(file.edit_users).to match_array [user.to_s]
         end
       end
     end
-  end
 
-  context "when read people change" do
-    let(:name) { 'abc@123.com' }
-    let(:type) { 'person' }
-    let(:access) { 'read' }
+    context "when edit groups change" do
+      let(:name) { 'my_edit_group' }
+      let(:type) { 'group' }
+      let(:access) { 'edit' }
 
-    it 'copies permissions to its contained files' do
-      # files have the depositor as the edit user to begin with
-      expect(work.file_sets.first.read_users).to eq []
+      it 'copies permissions to its contained files' do
+        # files have the depositor as the edit user to begin with
+        expect(work.file_sets.first.read_groups).to eq []
 
-      described_class.perform_now(work)
-      work.reload.file_sets.each do |file|
-        expect(file.read_users).to match_array ["abc@123.com"]
-        expect(file.edit_users).to match_array [user.to_s]
+        described_class.perform_now(work, use_valkyrie: false)
+        work.reload.file_sets.each do |file|
+          expect(file.edit_groups).to match_array ["my_edit_group"]
+          expect(file.edit_users).to match_array [user.to_s]
+        end
       end
     end
   end
 
-  context "when read groups change" do
-    let(:name) { 'my_read_group' }
-    let(:type) { 'group' }
-    let(:access) { 'read' }
+  context "when use_valkyrie is true" do
+    let(:resource) { valkyrie_create(:hyrax_work, edit_users: [user]) }
+    let(:file_set) { valkyrie_create(:hyrax_file_set) }
+    let(:user2) { create(:user) }
+
+    before do
+      resource.member_ids = Array(file_set.id)
+      file_set.permission_manager.acl.grant(:edit).to(user).save
+    end
+
+    context "when edit people change" do
+      it 'copies permissions to its contained files' do
+        resource.permission_manager.acl.grant(:edit).to(user2).save
+        expect(resource.edit_users).to match_array [user.to_s, user2.to_s]
+
+        # files have the depositor as the edit user to begin with
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
+          expect(file.edit_users).to match_array [user.to_s]
+        end
 
-    it 'copies permissions to its contained files' do
-      # files have the depositor as the edit user to begin with
-      expect(work.file_sets.first.read_groups).to eq []
+        described_class.perform_now(resource, use_valkyrie: true)
 
-      described_class.perform_now(work)
-      work.reload.file_sets.each do |file|
-        expect(file.read_groups).to match_array ["my_read_group"]
-        expect(file.edit_users).to match_array [user.to_s]
+        # files have both edit users from parent resource
+        Hyrax.query_service.custom_queries.find_child_filesets(resource: resource).each do |file|
```

Comment:
In reading this spec, it would be possible that we have an empty array, which would mean the spec is not testing anything.

It may be helpful to capture the `Hyrax.query_service.custom_queries.find_child_filesets(resource: resource)` as a variable and say that you expect it to have a specific count/size.